unity-renderer/Assets/DCLServices/ScreencaptureCamera/Service/ScreencaptureCameraService.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Cysharp.Threading.Tasks;
+using DCL;
+using DCL.Providers;
+using DCLFeatures.ScreencaptureCamera.CameraObject;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DCLServices.ScreencaptureCamera.Service
+{
+    public class ScreencaptureCameraService : IScreencaptureCameraService
+    {
+        private const string CONTROLLER_PATH = "ScreencaptureCameraController";
+        private const string MAIN_BUTTON_PATH = "ScreencaptureMainButton";
+
+        private readonly IAddressableResourceProvider resourceProvider;
+        private readonly BaseVariable<FeatureFlag> featureFlags;
+        private readonly DataStore_Player player;
+
+        private ScreencaptureCameraBehaviour cameraBehaviour;
+        private Canvas enableCameraButtonCanvas;
+
+        private bool featureIsEnabled => featureFlags.Get().IsFeatureEnabled("camera_reel");
+        private bool isGuest => UserProfileController.userProfilesCatalog.Get(player.ownPlayer.Get().id).isGuest;
```

Comment:
The correct way of doing this is injecting a `IUserProfileBridge` and then do: `userProfileBridge.GetOwn().isGuest`